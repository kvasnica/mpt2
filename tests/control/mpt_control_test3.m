function mpt_control_test3

% following changeset:
% http://control.ee.ethz.ch/~mpt/hg/mpt?cs=cedc93418d40
% introduced verification of sysStruct/probStruct with Options.useyalmip=1,
% which breaks input blocking features if mpt_optControl is used. fixed by:
% http://control.ee.ethz.ch/~mpt/hg/mpt?cs=1559dcb39dea

x0 = [1;1];
Double_Integrator
probStruct.Tconstraint = 0;
probStruct.Nc = 2;
probStruct.N = 3;

ctrl = mpt_control(sysStruct, probStruct);
u = mpt_getInput(ctrl, x0, struct('openloop', 1));
mbg_asserttolequal(u(2), u(3));
