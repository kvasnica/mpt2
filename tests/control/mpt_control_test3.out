No terminal weight "probStruct.P_N" specified... closed-loop stability will not be guaranteed
mpt_mpqp: 5 regions
-> Generated 1 partition.
Solution consists of 5 regions
